examples/generate-csr.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This script generates a simple SAN CSR to be used with Let's Encrypt
+# CA. Mostly intended for "auth --csr" testing, but, since it's easily
+# auditable, feel free to adjust it and use it on your production web
+# server.
+import os
+import argparse
+from OpenSSL import crypto
+
+try:
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.asymmetric import rsa, ec
+    from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, NoEncryption
+    cryptography=True
+except ImportError:
+    cryptography=False
+
+CWD = os.getcwd()
+
+
+def generate_parser():
```

Comment:
We really don't need a whole UI here. People who want an example of the openssl command to use can use the existing bash script; it's really quite sufficient to just have a small helper function that generates a csr in python.

Reply:
Bash script is not useful for Windows users. I took the occasion here to rewrite it to be cross-platform compatible, and so is meant to replace the existing bash script.

Following the existing logic in integration tests, it used by the tests to generate some CSR. You have some concern about it however. 

Do you want me to duplicate all this logic directly to misc when called for the tests?

Reply:
Postponed to next PR.